components/mediation/mediators/dataservices-mediator/org.wso2.micro.integrator.mediator.dataservice/src/main/java/org/wso2/micro/integrator/mediator/dataservice/DataServiceCallMediator.java

```diff
@@ -0,0 +1,390 @@
+        String propertyName = getPropertyName();
+        try {
+
+            OMElement omElement = DataServiceProcessor.dispatch(axis2MessageContext);
+
+            if (omElement != null) {
+                if (synLog.isTraceOrDebugEnabled()) {
+                    synLog.traceOrDebug("The result OMElement from the dataservice : " + omElement);
+                }
+
+                //set the result payload as property according to the target type
+                if (DataServiceCallMediatorConstants.TARGET_TYPE.equals(targetTypeName)) {
+                    messageContext.setProperty(propertyName, omElement);
+                    if (synLog.isTraceOrDebugEnabled()) {
+                        synLog.traceOrDebug("The result property : " + messageContext.
+                                getProperty(propertyName));
+                    }
+                } else {
+                    //	set the result payload as envelope in to message context according to the target type
+                    messageContext.getEnvelope().getBody().addChild(omElement);
+                }
+            }
+        } catch (DataServiceFault dataServiceFault) {
+            if (synLog.isTraceOrDebugEnabled()) {
+                synLog.traceOrDebug(dataServiceFault.getMessage());
+            }
+            synLog.error(dataServiceFault.getMessage());
+            handleException("DataService exception occurred while accessing the dataservice to do the operation",
+                    dataServiceFault, messageContext);
+
+        }
+    }
+
+    private void addOperations(Operations operations, OMElement payload, MessageContext msgCtx) {
+
+        for (Object operationObj : operations.getOperations()) {
+            if (operationObj instanceof Operation) {
+                Operation nestedOperation = (Operation) operationObj;
+                String operationName = nestedOperation.getOperationName();
+                OMElement nestedPayload = fac.createOMElement(operationName, omNamespace);
+                for (Param param : nestedOperation.getParams()) {
+                    addParams(param, nestedPayload, msgCtx);
+                }
+                payload.addChild(nestedPayload);
+            } else if (operationObj instanceof Operations) {
+                Operations rootOperations = (Operations) operationObj;
+                String rootOpName;
+                if (rootOperations.getType() == OperationsType.SINGLE_REQ) {
+                    Operation singleOp = (Operation) rootOperations.getOperations().get(0);
+                    rootOpName = singleOp.getOperationName();
+                } else {
+                    rootOpName = rootOperations.getType().toString().toLowerCase();
+                }
+                OMElement nestedOpEle = fac.createOMElement(rootOpName, omNamespace);
+                addOperations(rootOperations, nestedOpEle, msgCtx);
+                payload.addChild(nestedOpEle);
+            }
```

Comment:
else we need to throw unsupported type ?

Reply:
From the factory class it adds only the Operation and Operations objects. So no need to handle it from here.